File: LeetCode/1004/main.py
class Solution:
    def longestOnes(self, arr: List[int], k: int) -> int:
        n = len(arr)
        c, left, ans = 0, 0, 0

        for i in range(n):
            if arr[i] == 0:
                c += 1

            while c > k:
                if arr[left] == 0:
                    c -= 1
                left += 1

            ans = max(ans, i - left + 1)

        return ans

File: LeetCode/1004/main.cpp
class Solution {
public:
    int longestOnes(vector<int>& arr, int k) {
        int n = arr.size();
        int left = 0, c = 0, ans = 0;

        for(int i = 0; i < n; i++) {
            if(arr[i] == 0) {
                c++;
            } 

            while(c > k) {
                if(arr[left] == 0) {
                    c--;
                }
                left++;
            }

            ans = max(ans, i - left + 1);

        }

        return ans;

    }
};